src/renderer/react/token-box/index.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from 'react-redux';
+import store from '../common/store';
+import TokenBox from './containers/token-box';
+import TokenPrice from './containers/token-price';
+import { EthereumIcon, SelfkeyIcon, CustomIcon, CustomTokenText } from 'selfkey-ui';
+
+export class TokenBoxWrapper extends React.Component {
+	render() {
+		const {
+			cryptoCurrencyShort,
+			cryptoCurrencyName,
+			publicKey,
+			cryptoCurrency,
+			cryptoValue,
+			toCurrency,
+			toValue,
+			customTokenText,
+			copyAction,
+			transferAction
+		} = this.props;
+		console.log('PROPS', this.props);
+		return (
+			<Provider store={store}>
+				<TokenBox
+					cryptoCurrencyShort={cryptoCurrencyShort}
+					cryptoCurrencyName={cryptoCurrencyName}
+					publicKey={publicKey}
+					CryptoCurrencyIconComponent={
+						!cryptoCurrencyShort
```

Comment:
@rodrigopavezi maybe thous should be proper if else? Out of the jsx block

Reply:
I agree, however a helper function would be a better approach here.

Something like:

```
const renderIcon = (shortName) => {
  switch(shortName) {
    case 'KEY':
      return SelfkeyIcon;
    case 'ETH':
      return EthereumIcon;
    default:
      return: CustomIcon
  }
};